code/modules/mob/living/carbon/human/species_types/shadowpeople.dm

```diff
@@ -21,3 +22,62 @@
 			H.take_overall_damage(1,1)
 		else if (light_amount < 0.2) //heal in the dark
 			H.heal_overall_damage(1,1)
+
+
+/datum/species/shadow/nightmare
+	name = "Nightmare"
+	id = "nightmare"
+	limbs_id = "shadow"
+	burnmod = 1.5
+	blacklisted = TRUE
+	species_traits = list(NOBREATH,RESISTCOLD,RESISTPRESSURE,NOGUNS,NOBLOOD,RADIMMUNE,VIRUSIMMUNE,PIERCEIMMUNE,NODISMEMBER,NO_UNDERWEAR)
+	mutanteyes = /obj/item/organ/eyes/night_vision/nightmare
+
+/datum/species/shadow/nightmare/on_species_gain(mob/living/carbon/C, datum/species/old_species)
+	. = ..()
+	var/obj/effect/proc_holder/spell/targeted/shadowwalk/SW = new
+	C.AddSpell(SW)
+	var/obj/item/light_eater/blade = new
+	C.put_in_hands(blade)
+
+
+/datum/species/shadow/nightmare/bullet_act(obj/item/projectile/P, mob/living/carbon/human/H)
+	var/light_amount = 0
+	if(isturf(H.loc))
+		var/turf/T = H.loc
+		light_amount = T.get_lumcount()
+		if (light_amount < 0.2)
+			H.visible_message("<span class='danger'>[H] dances in the shadows, evading [P]!</span>")
+			playsound(T, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, 1)
+			return -1
+	return 0
+
+/obj/item/light_eater
+	name = "light eater"
+	icon_state = "arm_blade"
+	item_state = "arm_blade"
+	force = 25
+	lefthand_file = 'icons/mob/inhands/antag/changeling_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/antag/changeling_righthand.dmi'
+	flags_1 = ABSTRACT_1 | NODROP_1
+	w_class = WEIGHT_CLASS_HUGE
+	sharpness = IS_SHARP
+
+/obj/item/light_eater/afterattack(atom/movable/AM, mob/user, proximity)
+	if(!proximity)
+		return
+	if(isopenturf(AM)) //So you can actually melee with it
+		return
+	if(isliving(AM))
+		for(var/obj/item/O in AM)

```

Comment:
I think you just want /obj

Reply:
I dont want them disintegrating machines and structures in a single hit